app/models/manageiq/consumption/showback_event.rb

```diff
@@ -41,13 +41,13 @@ def validate_format
     nil
   end
 
-  def generate_data
+  def generate_data(data_units = ManageIQ::Consumption::ConsumptionManager.load_column_units)
     self.data = {}
     ManageIQ::Consumption::ShowbackUsageType.all.each do |measure_type|
       next unless resource_type.include?(measure_type.category)
       self.data[measure_type.measure] = {}
       measure_type.dimensions.each do |dim|
-        self.data[measure_type.measure][dim] = 0 unless measure_type.measure == "FLAVOR"
+        self.data[measure_type.measure][dim] = [0,data_units[dim.to_sym] || "" ] unless measure_type.measure == "FLAVOR"
```

Comment:
What happens if measure = FLAVOR?. We don't create the structure?

Reply:
In the generation of the data not, That will be create in update_event. Any way we can initialize this object in generate_data too. I'll make the changes